src/Avalonia.Visuals/Animation/Animators/BaseBrushAnimator.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Disposables;
+using Avalonia.Logging;
+using Avalonia.Media;
+
+namespace Avalonia.Animation.Animators
+{
+    /// <summary>
+    /// Animator that handles all animations on properties
+    /// with <see cref="IBrush"/> as their type and
+    /// redirect them to the properly registered
+    /// animators in this class.
+    /// </summary>
+    public class BaseBrushAnimator : Animator<IBrush>
+    {
+        private IAnimator _targetAnimator;
+
+        private static readonly List<(Func<Type, bool> Match, Type AnimatorType)> _brushAnimators = new List<(Func<Type, bool> Match, Type AnimatorType)>();
+
+        /// <summary>
+        /// Register an <see cref="Animator{T}"/> that handles a specific
+        /// <see cref="IBrush"/>'s descendant value type.
+        /// </summary>
+        /// <param name="condition">
+        /// The condition to which the <see cref="Animator{T}"/>
+        /// is to be activated and used.
+        /// </param>
+        /// <typeparam name="TAnimator">
+        /// The type of the animator to instantiate.
+        /// </typeparam>
+        public static void RegisterBrushAnimator<TAnimator>(Func<Type, bool> condition)
+            where TAnimator : IAnimator
+        {
+            _brushAnimators.Insert(0, (condition, typeof(TAnimator)));
+        }
+        
+        /// <inheritdoc/>
+        public override IDisposable Apply(Animation animation, Animatable control, IClock clock,
+            IObservable<bool> match, Action onComplete)
+        {
+            foreach (var valueType in _brushAnimators
+                .Where(valueType => valueType.Match(this[0].Value.GetType())))
```

Comment:
In this case Linq shouldn't cause any performance problems, but it should be quite easy to avoid it here anyway.

Reply:
idk, it wasnt linq at all but rider kept giving me warning to turn it into a linq query.

Reply:
@jmacato remove the Where... might be nice to aim for a Linq free codebase

Reply:
a simple if will be better